src/labCommands.ts

```diff
@@ -143,6 +143,29 @@ export function addJLabCommands(
       },
       isEnabled
     }),
+    commands.addCommand('vim:leave-current-mode', {
+        label: 'Move Insert to Normal to Command Mode"',
+        execute: args => {
+            const current = getCurrent(args);
+
+            if (current) {
+                const { content } = current;
+                if (content.activeCell !== null) {
+                    let editor = content.activeCell.editor as CodeMirrorEditor;
+
+                    // Get the current editor state
+                    if(editor.editor.state.vim.insertMode) {
+                      (CodeMirror as any).Vim.handleKey(editor.editor, '<Esc>');
+                    } else if (editor.editor.state.vim.visualMode){
+                      (CodeMirror as any).Vim.handleKey(editor.editor, '<Esc>');
```

Comment:
You can these two ifs as they result in the same command being executed.

```suggestion
                    if(editor.editor.state.vim.insertMode || editor.editor.state.vim.visualMode) {
                      (CodeMirror as any).Vim.handleKey(editor.editor, '<Esc>');
```